src/rust/engine/remote_provider/remote_provider_reapi/src/byte_store.rs

```diff
@@ -390,6 +392,28 @@ impl ByteStoreProvider for Provider {
         .map_err(|e| e.to_string())
     }
 
+    fn batch_load_supported(&self) -> bool {
+        self.batch_load_enabled
+    }
+
+    async fn load_batch(&self, digests: Vec<Digest>) -> Result<Vec<Bytes>, String> {
```

Comment:
I think the response type needs to be tweaked here still. Individual digests can error in the batch, something like `Result<Vec<Result<Bytes, String>, String>>`